docs/examples/getting_started/network.tf

```diff
@@ -0,0 +1,123 @@
+# Creating a Virtual Cloud Network (VCN)
+resource "oci_core_virtual_network" "MyVcn" {
+  cidr_block = "${var.VPC-CIDR}"
+  dns_label = "myvcn"
+  compartment_id = "${var.compartment_ocid}"
+  display_name = "MyVcn"
+}
+
+# Setting up an Internet Gateway.
+# This is used to send-receive traffic between the Internet and (compute) instances hosted inside our VCN.
+resource "oci_core_internet_gateway" "MyInternetGateway" {
+  compartment_id = "${var.compartment_ocid}"
+  display_name = "MyInternetGateway"
+  vcn_id = "${oci_core_virtual_network.MyVcn.id}"
+}
+
+# Setting up a Route Table.
+# Network traffic intended to head out to the general Internet (0.0.0.0/0) is being routed to our Internet Gateway.
+resource "oci_core_route_table" "MyRouteTable" {
+  compartment_id = "${var.compartment_ocid}"
+  vcn_id = "${oci_core_virtual_network.MyVcn.id}"
+  display_name = "MyRouteTable"
+  route_rules {
+    cidr_block = "0.0.0.0/0"
+    network_entity_id = "${oci_core_internet_gateway.MyInternetGateway.id}"
+  }
+}
+
+# Setting up Security Lists.
+# The allow/deny rules in the security list determine if a subnet is public or private.
+resource "oci_core_security_list" "MyPublicSubnetSecurityList" {
+  compartment_id = "${var.compartment_ocid}"
+  display_name = "MyPublicSubnetSecurityList"
+  vcn_id = "${oci_core_virtual_network.MyVcn.id}"
+  egress_security_rules = [{
+    destination = "0.0.0.0/0"
+    protocol = "6"
+  }]
+  ingress_security_rules = [{
+    tcp_options {
+      "max" = 80
+      "min" = 80
+    }
+    protocol = "6"
+    source = "0.0.0.0/0"
+  },
+    {
+      protocol = "6"
+      source = "${var.VPC-CIDR}"
+    }]
+}
+
+# The allow/deny rules in the security list determine if a subnet is public or private.
+# If we choose to run Oracle RDBMS in our private subnet, we might want to allow traffic to port 1521 etc.
+resource "oci_core_security_list" "MyPrivateSubnetSecurityList" {
+  compartment_id = "${var.compartment_ocid}"
+  display_name = "MyPrivateSubnetSecurityList"
+  vcn_id = "${oci_core_virtual_network.MyVcn.id}"
+  egress_security_rules = [{
+    protocol = "6"
+    destination = "${var.VPC-CIDR}"
+  }]
+  ingress_security_rules = [{
+    protocol = "6"
+    source = "${var.VPC-CIDR}"
+  }]
+}
+
+# Creating a public subnet within our VCN.
+# Instances hosted in public-subnets can be accessed from the Internet.
+resource "oci_core_subnet" "MyPublicSubnetAD1" {
+  availability_domain = "${lookup(data.oci_identity_availability_domains.ADs.availability_domains[0], "name")}"
+  cidr_block = "10.0.1.0/24"
```

Comment:
If we parameterize the VCN CIDR, then we should calculate the subnet CIDRs based on that using the cidrsubnet function - see https://www.terraform.io/docs/configuration/interpolation.html#cidrsubnet-iprange-newbits-netnum-